### PR TITLE
test(platform-server): add SSR integration test for null and normal input values

### LINKED ISSUE #69785: SSR: Input value null is seralized as null string.
`<input [value]="null">` is being serialized as `<input value="null">` resulting into a flickering during hydration. (shows input shows `null` before being hydrated and becoming empty. 

Related #58780

### PR DESCRIPTION
**Pull request by surajy93**
test(platform-server): add SSR integration test for null and normal input values

Fixes #69785

Add an Angular SSR integration test in platform-server verifying that null input values do not render string attributes like value="null" during server-side rendering, while normal non-empty string values like value="hello" are properly preserved.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/contributing-docs/commit-message-guidelines.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.dev application / infrastructure changes
- [x] Other... Please describe: Integration Test

## What is the current behavior?

Issue Number: #69785

During Angular Server-Side Rendering (SSR), binding `null` properties such as `<input [value]="null">` serializes string attributes like `<input value="null">`, which causes a hydration flicker when the client app hydrates.

## What is the new behavior?

Adds an Angular SSR integration test in `platform-server` (`full_app_hydration_spec.ts`) verifying that:
1. `null` input values omit string attributes like `value="null"`.
2. Normal non-empty string values such as `value="hello"` are properly preserved.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information
This PR is kept as pending merge of Domino PR [angular/domino#34](https://github.com/angular/domino/pull/34) and a subsequent Domino version bump in `package.json`.

image

### CODE REVIEW

File: packages/platform-server/test/full_app_hydration_spec.ts [hunk 1]
```diff
@@ -211,6 +211,26 @@ describe('platform-server full application hydration integration', () => {
...
+        })
+        class AppComponent {
+          nullValue: string | null = null;
+          normalValue = 'hello';
+        }
+
+        const html = await ssr(AppComponent);
+        const ssrContents = getAppContents(html);
+
+        expect(ssrContents).not.toContain('value="null"');
```
Comment (SkyZeroZx) [inline packages/platform-server/test/full_app_hydration_spec.ts:231]: We should probably improve these expects; we might want to validate the HTML output for each input case as it should be.

Comment (surajy93) [reply packages/platform-server/test/full_app_hydration_spec.ts:231]: Yes agreed will do once pending PR  https://github.com/angular/domino/pull/34 get merged

**Review by JeanMeche — CHANGES_REQUESTED** [review]
You'll first need to update the sha in the package.json to use the latest domino

**Comment by surajy93** [comment]
> You'll first need to update the sha in the package.json to use the latest domino

update the sha please review again thanks @JeanMeche

**Review by JeanMeche — APPROVED** [review]

**Review by alan-agius4 — APPROVED** [review]

**Comment by thePunderWoman** [comment]
@surajy93 This has merge conflicts and will need to be cleaned up before we can land it.

**Comment by surajy93** [comment]
> @surajy93 This has merge conflicts and will need to be cleaned up before we can land it.
clean up done @thePunderWoman

**Comment by thePunderWoman** [comment]
@surajy93 Looks like you have a merge commit though. Can you rebase and squash that?

**Comment by surajy93** [comment]
> @surajy93 Looks like you have a merge commit though. Can you rebase and squash that?

@thePunderWoman done

**Comment by thePunderWoman** [comment]
@surajy93 How dare you be so quick and responsive! :) Thank you.

**Comment by surajy93** [comment]
> @surajy93 How dare you be so quick and responsive! :) Thank you.

haha just to beat AI :q thanks for review

**Comment by thePunderWoman** [comment]
This PR was merged into the repository. The changes were merged into the following branches:

- main: fb698d12cdda1b626d727b35e57411b2a7b75336
- 22.1.x: 832ff2aff3148b09126dd0c28672652d2b21899a
